MissionTable.cpp

```diff
@@ -115,32 +134,16 @@ void WriteBoolCell(bool boolean, std::streamsize cellWidthInChars)
 
 
 
-std::string ConvertMissionTypeToString(MissionTypes missionType)
+std::string_view ConvertMissionTypeToString(MissionTypes missionType)
 {
-	switch (missionType)
-	{
-	case Colony:
-		return "Colony Game";
-	case AutoDemo:
-		return "Demo";
-	case Tutorial:
-		return "Tutorial";
-	case MultiLandRush:
-		return "Land Rush";
-	case MultiSpaceRace:
-		return "Space Race";
-	case MultiResourceRace:
-		return "Resource Race";
-	case MultiMidas:
-		return "Midas";
-	case MultiLastOneStanding:
-		return "Last One Standing";
-	default:
-		// Positive missionTypes represent campaign mission index.
-		if (static_cast<int>(missionType) > 0) {
-			return "Campaign";
-		}
-
-		throw std::runtime_error("An improper MissionType enum value of " + std::to_string(missionType) + " was provided.");
+	// Positive missionTypes represent campaign mission index.
+	if (static_cast<int>(missionType) > 0) {
+		return "Campaign";
 	}
+	auto missionTypeIndex = static_cast<std::size_t>(-missionType) - 1;
```

Comment:
Probably worth a comment here explaining why missionType is being multiplied by negative one just to be cast to a size_t.

Reply:
Good point.